runtime/lua/vim/filetype.lua

```diff
@@ -1175,6 +1241,67 @@ local extension = {
       return 'rc'
     end
   end,
+  control = function(path, bufnr)
+    return require('vim.filetype.detect').control(bufnr)
+  end,
+  copyright = function(path, bufnr)
+    return require('vim.filetype.detect').copyright(bufnr)
+  end,
+  -- Ignored extensions
+  bak = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  ['dpkg-dist'] = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  ['dpkg-old'] = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  ['dpkg-new'] = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  ['dpkg-bak'] = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  new = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  old = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  orig = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  pacsave = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  pacnew = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  rpmsave = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  rmpnew = function(path, bufnr)
+    local root = vim.fn.fnamemodify(path, ':r')
+    return M.match(root, bufnr)
+  end,
+  ['in'] = function(path, bufnr)
+    if vim.fn.expand(path, ':t') ~= 'configure.in' then
+      local root = vim.fn.fnameescape(vim.fn.fnamemodify(path, ':r'))
```

Comment:
Using `fnameescape` here, but not in the other patterns above. I wonder if we need it at all? My guess is that we do not since we are just passing the result of `fnamemodify` directly to `vim.filetype.match`.

Reply:
I've removed it for now